src/scripts/safari/Safari.ts

```diff
@@ -487,8 +492,19 @@ class Safari {
         return false;
     }
 
+    static safariProgressTooltip() {
+        const tooltip = { trigger : 'hover', title : '' };
+        const level = Safari.safariLevel();
+        if (level == Safari.maxSafariLevel) {
+            tooltip.title = 'Max level reached';
+        } else {
+            tooltip.title = `${Safari.safariExp() - Safari.expRequiredForLevel(level)} / ${Safari.expRequiredForLevel(level + 1) - Safari.expRequiredForLevel(level)}`;
+        }
+        return tooltip;
+    }
+
     private static expRequiredForLevel(level: number) {
-        return Math.ceil(2000 * (1.2 ** (level - 1) - 1));
+        return Math.ceil(2000 * (1.2 ** (Math.min(Safari.maxSafariLevel, level) - 1) - 1));
```

Comment:
Is this change needed?

Reply:
No. I tried to fulfill the xp bar that way before changing my mind.
Back to normal.